liblustreapi/src/lib.rs

```diff
@@ -86,11 +86,11 @@ pub fn fid2path(device: &str, fidstr: &str) -> Result<String, LiblustreError> {
     let mut buf: Vec<u8> = vec![0; std::mem::size_of::<u8>() * PATH_BYTES];
     let ptr = buf.as_mut_ptr() as *mut libc::c_char;
 
-    let devptr = device.as_ptr() as *const i8;
-    let fidptr = fidstr.as_ptr() as *const i8;
+    let devptr = CString::new(device)?.into_raw();
+    let fidptr = CString::new(fidstr)?.into_raw();
```

Comment:
You'll want to make sure you call `from_raw` after the ffi call:

https://doc.rust-lang.org/std/ffi/struct.CString.html#method.into_raw